tests/components/template/test_sensor.py

```diff
@@ -438,6 +442,39 @@ def test_available_template_with_entities(self):
         )
 
 
+async def test_creating_sensor_loads_group(hass):
+    """Test setting up template sensor loads group component first."""
+    order = []
+
+    async def async_setup_group(hass, config):
+        # Make sure group takes longer to load, so that it won't
+        # be loaded first by chance
+        await sleep(0.1)
+
```

Comment:
We shouldn't sleep here as it won't be reliable.

Reply:
Right, I'll use an event.